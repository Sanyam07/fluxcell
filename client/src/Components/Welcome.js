import React from 'react';
import {
  Card,
  CardImg,
  CardText,
  CardBody,
  CardTitle,
  CardSubtitle,
  Button,
} from 'reactstrap';
import styled from 'styled-components';

export default function Welcome() {

  const StyledCard = styled(Card)`
      background-color: white;
      &:hover {
        background-color: #caffe5;
      }
    `;

  return (
    <StyledCard>
      <CardImg
        top
        width="100%"
        src="https://i.imgur.com/n744BL9.png"
        alt="Card image cap"
      />
      <CardBody>
        <CardTitle>Welcome to Fluxdash</CardTitle>
        <CardSubtitle>Will be awesome</CardSubtitle>
        <CardText />
        <Button color="primary">This button does nothing</Button>
      </CardBody>
    </StyledCard>
  );
}
