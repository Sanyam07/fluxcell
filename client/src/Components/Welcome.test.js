import React from 'react';
import renderer from 'react-test-renderer';
import Welcome from './Welcome';

test('Link changes the class when hovered', () => {
  const component = renderer.create(
    <Welcome />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseEnter();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseLeave();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
