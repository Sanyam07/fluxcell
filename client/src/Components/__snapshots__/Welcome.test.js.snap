// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<div
  className="sc-bdVaJa hqgvtt card"
>
  <img
    alt="Card image cap"
    className="card-img-top"
    src="https://i.imgur.com/n744BL9.png"
    width="100%"
  />
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      Welcome Fluxcraft GUI
    </h5>
    <h6
      className="card-subtitle"
    >
      Will be awesome
    </h6>
    <p
      className="card-text"
    />
    <button
      className="btn btn-primary"
      onClick={[Function]}
      type={undefined}
    >
      This button does nothing
    </button>
  </div>
</div>
`;
